### PR TITLE
Add per train route information while selecting routes

### LINKED ISSUE #98: Running trains enhancements
Would be great to create a little train card that displays:

Train number
Train status (has/hasn't run)
Train income (if it has run, how much did the route earn)

Optional:
Train image (just a little visual enhancement), I'd keep it to a square icon like the company icons for now.

### PR DESCRIPTION
**Pull request by jenf**
Give more information about the route while the user is selecting it
Also makes the route colours colourblind safe

Valid route example

Screenshot from 2020-05-05 22-24-42

Invalid route example
Screenshot from 2020-05-05 22-25-00

fixes #98

### CODE REVIEW

File: assets/app/view/route_selector.rb [hunk 1]
```diff
@@ -10,6 +10,13 @@ class RouteSelector < Snabberb::Component
     needs :routes, store: true, default: []
     needs :selected_route, store: true, default: nil
 
+    # Get routes that have a length greater than zero
+    # Due to the way this and the map hook up routes needs to have
+    # an entry, but that route is not valid at zero length
+    def active_routes
+      @routes.map { |i| i unless i.hexes.empty? }.compact
```
Comment (tobymao) [inline assets/app/view/route_selector.rb:17]: @routes.select { |r| r.hexes.any? }

File: assets/app/view/route_selector.rb [hunk 2]
```diff
@@ -28,36 +35,75 @@ def render
...
-          cursor: selected ? 'none' : 'pointer',
+          cursor: selected ? 'default' : 'pointer',
           margin: '0.5rem 0.5rem 0.5rem 0',
           padding: '0.5rem',
         }
 
-        h(:div, { style: style, on: { click: onclick } }, "Train: #{train.name}")
+        route = @routes.find { |t| t.train == train }
+        children = []
+        if route && !route.hexes.empty?
```
Comment (tobymao) [inline assets/app/view/route_selector.rb:45]: if (route = active_routes.find { |t| t.tarin == train })

File: assets/app/view/route_selector.rb [hunk 3]
```diff
@@ -28,36 +35,75 @@ def render
...
+          revenue, invalid = begin
+                               [@game.format_currency(route.revenue), nil]
+                             rescue Engine::GameError => e
```
Comment (tobymao) [inline assets/app/view/route_selector.rb:48]: why would there be a game error

File: assets/app/view/route_selector.rb [hunk 4]
```diff
@@ -28,36 +35,75 @@ def render
...
+                               ['N/A', e.to_s]
+                             end
+
+          style['background-color'] = Part::Track::ROUTE_COLORS[@routes.index(route)]
+          style['color'] = 'white'
+          children << h(:td, route.stops.length)
```
Comment (tobymao) [inline assets/app/view/route_selector.rb:54]: always use size not length

**Comment by gchambers419** [comment]
Thanks for the colorblind fix - the old colors were marginally usable for someone with my type of colorblindness.  If you update your test to include the fourth train color I can bless that one too.  (New to 18xx are ther any games that allow five trains?)

The selected and error highlights could use just a bit more emphasis from my perspective.  Maybe a little wider, or as a ring around the box, separated by a white gap?

File: assets/app/view/route_selector.rb (hunk 3 above)
Comment (jenf) [reply assets/app/view/route_selector.rb:48]: route.revenue throws an exception if it can't run (to long a route etc)

**Comment by jenf** [comment]
> Thanks for the colorblind fix - the old colors were marginally usable for someone with my type of colorblindness. If you update your test to include the fourth train color I can bless that one too. (New to 18xx are ther any games that allow five trains?)
Technically that color pallette goes to 13, there's 4 at the moment.
> 
> The selected and error highlights could use just a bit more emphasis from my perspective. Maybe a little wider, or as a ring around the box, separated by a white gap?

Added another pixel. Maybe will do further improvements on a separate pr.

**Pull request merged**